components/javascript-wrapper.js

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * This file contacts the common code used by all of the variations of any RDF
+ * node componentin the noflo implementation of the RDF Pipeline.
+ */
+
+var _ = require('underscore');
+var stateComponent = require('./state-component');
+
+/**
+ * This module provides a Javascript wrapper for updaters.  It is responsible for setting up the
+ * code to ensure updater callback, and then invoking the RDF pipeline framework to complete building
+ * the component and setting up the vni and state metadata.
+ *
+ * @param def updater definition. This can be a simple updater function, or an object that contains 
+ *        updater metadata (e.g., input/output port definitions, description, icons), and an updater.
+ *
+ * @return a promise to create the noflo rdf component
+ */
+module.exports = function(def) { 
+
+    var self = this;
+
+    if ( _.isUndefined(def) || ( _.isEmpty(def) && ! _.isFunction( def ))) { 
+        // Nothing to work with - no metadata or updater - can't proceed.
+        throw new Error('javascript-wrapper found no updater definition');
```

Comment:
OK - will add the defaults for the next check in.

The javascript-wrapper does not know how many connections we have - it only has the port definitions which are not even instantiated yet at the time we see it - that policy enforcement needs to be lower.